xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-ui/src/main/resources/XWiki/XWikiUserSheet.xml

```diff
@@ -46,38 +46,21 @@
 ## Setting categories
 #########################
 #set($categories = [])
-#set($discard = $categories.add({'id':'profile', 'sheet':'XWiki.XWikiUserProfileSheet', 'glyphicon': 'user'}))
-#set($isMyProfile = ($services.model.resolveDocument($xcontext.user) == $doc.documentReference))
-#if($isMyProfile || $hasAdmin)
-  #set($discard = $categories.add({'id':'preferences', 'sheet':'XWiki.XWikiUserPreferencesSheet', 'glyphicon': 'wrench'}))
-#end
-#set($discard = $categories.add({'id':'groups', 'sheet':'XWiki.XWikiUserMembershipSheet', 'glyphicon': 'group'}))
-## TODO: add APIs to be able to display users watchlists to admins
-#if($isMyProfile &amp;&amp; $hasWatch)
-  #set($discard = $categories.add({'id':'watchlist', 'sheet':'XWiki.XWikiUserWatchListSheet', 'glyphicon': 'eye'}))
-#end
-#if($isMyProfile)
-  #if ("$!services.notification.watch" != '' || $hasWatch)
-    #set($discard = $categories.add({'id':'network', 'sheet':'XWiki.XWikiUserNetworkSheet', 'glyphicon': 'world'}))
-  #end
-  ## TODO: add an enhancement system instead
-  #set($notificationsSheet = $services.model.createDocumentReference('', ['XWiki', 'Notifications', 'Code'], 'XWikiUserNotificationsSheet'))
-  #if ($xwiki.exists($notificationsSheet))
-    #set($discard = $categories.add({'id':'notifications', 'sheet':'XWiki.Notifications.Code.XWikiUserNotificationsSheet', 'glyphicon': 'bell'}))
+
+## load the user menu from the 'org.xwiki.plaftorm.user.profile.menu' UIXP.
+#foreach ($uix in $services.uix.getExtensions('org.xwiki.plaftorm.user.profile.menu', {'sortByParameter': 'priority'}))
+  #if($uix.parameters.get('isActive') == 'true')
```

Comment:
I'm wondering if it wouldn't be safer to do the contrary: only hide the UIX if `isActive == 'false'`. By definition, the UIX should be injected here we added a parameter to control it, but the fallback should be to view them if the parameter is missing.
I think it would be better especially if we're gonna get rid of this parameter in the future, and transform it to a UIX class parameter: this would allow to create new UIX without needed to remove it. WDYT?

Reply:
I find it more interesting and to be strict in the displayed entries as it is undesirable to display entries to an unauthorized user.
On the cons part, the current behaviour has more chances to lead to blocking bug.
I'm voting +0 on the change to `isActive == false`.